### PR TITLE
Unify and Integrate Robot Constants with Dependency Injection

### LINKED ISSUE #2066: Create a `RobotConstants` struct to Initialize `FirmwareRobot`s
### Description of the task
Robot dependent constants like max_allowable_linear_acceleration, max_allowable_linear_speed, max_allowable_angular_acceleration, and max_allowable_angular_speed, JERK_LIMIT, WHEEL_MOTOR_PHASE_RESISTANCE, etc. are robot version dependent, so we need to be able to set these independently for each robot. This struct should be passed in to the constructor of FirmwareRobot and the values should replace uses of the constants file

### Acceptance criteria

- [ ] Create a `RobotConstants` struct
- [ ] initialize `FirmwareRobot` with this struct
- [ ] remove the associated constants from the constants file

### PR DESCRIPTION
**Pull request by jonathanlew**
### Description
* created RobotConstants struct with creator method
* created WheelConstants struct with creator method
* removed constants from constants.h and dependency injected the structs into software and firmware
* added RobotConstants to the Robot class

### Testing Done
added more unit tests on sharedphysics conversion function

### Resolved Issues
resolves #2066

### Length Justification
:(

### Review Checklist

**_It is the reviewers responsibility to also make sure every item here has been covered_**

- [ ] **Function & Class comments**: All function definitions (usually in the `.h` file) should have a javadoc style comment at the start of them. For examples, see the functions defined in `thunderbots/software/geom`. Similarly, all classes should have an associated Javadoc comment explaining the purpose of the class.
- [ ] **Remove all commented out code**
- [ ] **Remove extra print statements**: for example, those just used for testing
- [ ] **Resolve all TODO's**: All `TODO` (or similar) statements should either be completed or associated with a github issue
- [ ] **Justify drops in code coverage**: If this PR results in a non-trivial drop in code coverage (a bot should post a coverage diagram as a comment), please justify why we can't test the code that's not covered.

### CODE REVIEW

**Review by itsarune — COMMENTED** [review]
great 👍 👍, some minor things

File: src/firmware/app/world/firmware_robot_test.cpp [hunk 1]
```diff
@@ -158,10 +155,9 @@ TEST_F(FirmwareRobotTest, getRobotConstants)
 {
     RobotConstants_t constants = app_firmware_robot_getRobotConstants(firmware_robot);
 
-    EXPECT_NEAR(1.1, constants.mass, 1e-5);
-    EXPECT_NEAR(1.2, constants.moment_of_inertia, 1e-5);
-    EXPECT_NEAR(1.3, constants.robot_radius, 1e-5);
-    EXPECT_NEAR(1.4, constants.jerk_limit, 1e-5);
+    EXPECT_NEAR(1.1, constants.mass_kg, 1e-5);
+    EXPECT_NEAR(1.2, constants.moment_of_inertia_kg_m_2, 1e-5);
+    EXPECT_NEAR(1.4, constants.jerk_limit_kg_m_per_s_3, 1e-5);
```
Comment (tarakong01) [inline src/firmware/app/world/firmware_robot_test.cpp]: Just confirming if you purposely omitted the robot_radius test here?

Comment (jonathanlew) [reply src/firmware/app/world/firmware_robot_test.cpp]: not on purpose, fixed

File: src/software/simulation/force_wheel_simulator_robot_singleton.h [hunk 2]
```diff
@@ -47,11 +47,12 @@ class ForceWheelSimulatorRobotSingleton : public SimulatorRobotSingleton
      * calling the given function. If the simulator_robot is invalid, a warning is logged
      * and a default value is returned.
      *
+     * @tparam RET_VAL the return value of the function to execute
```
[the remark this answers is not in the document]
Comment (jonathanlew) [reply src/software/simulation/force_wheel_simulator_robot_singleton.h:50]: it's a type param for the template

File: src/software/simulated_tests/terminating_validation_functions/robot_state_validation.cpp [hunk 3]
```diff
@@ -35,26 +35,26 @@ void robotAtOrientation(RobotId robot_id, std::shared_ptr<World> world_ptr,
...
+        world_ptr->friendlyTeam().getRobotById(robot_id);
+    if (!robot_optional.has_value())
+    {
+        LOG(FATAL) << "There is no robot with ID: " + std::to_string(robot_id);
+    }
+
+    while ((robot_optional->orientation().minDiff(orientation) >
+            close_to_orientation_threshold))
+    {
+        robot_optional = world_ptr->friendlyTeam().getRobotById(robot_id);
```
[the remark this answers is not in the document]
Comment (jonathanlew) [reply src/software/simulated_tests/terminating_validation_functions/robot_state_validation.cpp]: yeah it was, but it's duplicated code, so I cleaned it up

**Review by itsarune — APPROVED** [review]
👍

**Review by tarakong01 — APPROVED** [review]
looks good!

**Review by akhilveeraghanta — APPROVED** [review]

**Pull request merged**